EnclaveWrapper/EnclaveWrapper/SgxException.cpp

```diff
@@ -0,0 +1,19 @@
+#include "stdafx.h"
+#include "SgxException.h"
+
+#define BUFSIZE 256
+
+SgxException::SgxException(sgx_status_t errorCode) {
+	this->errorCode = errorCode;
+	buf = new char[BUFSIZE];
```

Comment:
That buffer is only used in `what` method so maybe it would be better to allocate that memory there and here you could just set it to `null`. Having said all this I think that fixed buffer size causes only more troubles when working with strings. Please consider using [std::string](http://www.cplusplus.com/reference/string/string/) instead.

Reply:
IMO it would be unnecessarily complex since I need to combine hex value with a string (researched that a bit and it doesn't seem that there is a super simple solution for that with std::string) , and after that we would need to convert that to char* anyway.

Reply:
Do you mean the `sprintf` part? Well, it's not that hard:
```
#include <sstream>

...

  std::stringstream stream;
  stream << "SGX Exception, error code was " << std::hex << this->errorCode << std::endl;
  std::string result( stream.str() );
```

We need to "convert" it to `const char*` as you said but that's ok. What `std::string` gives us is the ability to dynamically reallocate buffer when it's needed. We wouldn't need to handle it by ourselves.